script.js

```diff
@@ -1,38 +1,69 @@
 const state = {
+  shows: [],
+  selectedShowId: "",
   episodes: [],
   searchTerm: "",
   selectedEpisodeId: "",
 };
 
-function fetchEpisodes(){
-  return fetch("https://api.tvmaze.com/shows/82/episodes").then(function(data){
-    if(data.ok){
-     return data.json();
-    }
-    throw new Error("Something went wrong");
-  });
+async function fetchAllShows() {
+  return await fetch("https://api.tvmaze.com/shows")
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Failed to fetch shows");
+      }
+      return response.json();
+    })
+    .then((data) => {
+      state.shows = data;
+      state.shows.sort(function (a, b) {
+        return a.name.toLowerCase().localeCompare(b.name.toLowerCase());
+      });
+      state.selectedShowId = state.shows[0].id;
+      fetchEpisodes();
+      render();
+      return data;
+    })
+    .catch((error) => {
+      console.error(error);
+      throw error;
+    });
 }
 
-function render() {
-  fillEpisodeList();
-  renderBySelect();
-  renderBySearch();
+async function fetchEpisodes() {
+  const showId = state.selectedShowId;
+  return await fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Failed to fetch episodes");
+      }
+      return response.json();
+    })
+    .then((data) => {
+      state.episodes = data;
+      render();
+      console.log(data);
+      return data;
+    })
+    .catch((error) => {
+      console.error(error);
+      throw error;
+    });
 }
 
 setTimeout(function timeout() {
```

Comment:
You should not be using set timeout to fetch data you should just use async and a wait that will do the job for you!

Reply:
I will update this, it was added by Rabia but I understand your explanation. Thank you

Reply:
Ive put this for testing, slow response from the API.